### ISSUE TITLE
[pytorch] bug ModuleListImpl ModuleDictImpl SequentialImpl these class can not use insert or put  method ,will crash the process

### ISSUE THREAD

**Issue by mullerhai** [bytedeco/javacpp-presets#1299]
Hi, I want to organize the use defined stack module layer use module vector or dict to do that ,
In python:
```
class MultiLayerPerceptron(torch.nn.Module):

    def __init__(self, input_dim, embed_dims, dropout, output_layer=True):
        super().__init__()
        layers = list()
        for embed_dim in embed_dims:
            layers.append(torch.nn.Linear(input_dim, embed_dim))
            layers.append(torch.nn.BatchNorm1d(embed_dim))
            layers.append(torch.nn.ReLU())
            layers.append(torch.nn.Dropout(p=dropout))
            input_dim = embed_dim
        if output_layer:
            layers.append(torch.nn.Linear(input_dim, 1))
        self.mlp = torch.nn.Sequential(*layers)

    def forward(self, x):
        """
        :param x: Float tensor of size ``(batch_size, embed_dim)``
        """
        return self.mlp(x)

```

in scala  first of all I create  linearBnReluDropBlock  ,then create  MultiLayerPerceptron

```
import org.bytedeco.javacpp._
import org.bytedeco.pytorch.global.torch
import org.bytedeco.pytorch.global.torch.ScalarType
import org.bytedeco.pytorch.{BatchNorm2dImpl, BatchNormOptions, Conv1dImpl, Conv1dOptions, Conv2dImpl, Conv2dOptions, Module, ReLUImpl, ReLUOptions, Tensor, _}

import scala.collection.mutable.{ArrayBuffer, ListBuffer}
import scala.language.postfixOps

class LinearBnReluDropBlock(val input_dim: Long, val embed_dim: Long, val dropoutPer: Double, val step: String) extends Module {

  val linear: LinearImpl = new LinearImpl(input_dim, embed_dim)
  register_module(s"linear_${step}", linear)
  val bnOpt = new BatchNormOptions(embed_dim)
  var bn: BatchNorm2dImpl = new BatchNorm2dImpl(bnOpt)
  register_module(s"bn_${step}", bn)
  val reluOpt = new ReLUOptions
  reluOpt.inplace.put(true)
  val relu = new ReLUImpl(reluOpt)
  register_module(s"relu_${step}", relu)
  val dropOpt = new DropoutOptions(dropoutPer)
  val dropout = new DropoutImpl(dropOpt)
  register_module(s"drop_${step}", dropout)

  def forward(x: Tensor): Tensor = dropout.forward(relu.forward(bn.forward(linear.forward(x))))
}

```

```
package org.pytorch.layer

import org.bytedeco.javacpp._
import org.bytedeco.pytorch.global.torch
import org.bytedeco.pytorch.global.torch.ScalarType
import org.bytedeco.pytorch.{BatchNorm2dImpl, BatchNormOptions, Conv1dImpl, Conv1dOptions, Conv2dImpl, Conv2dOptions, Module, ReLUImpl, ReLUOptions, Tensor, _}

import scala.collection.mutable.{ArrayBuffer, ListBuffer}
import scala.language.postfixOps

/** *
 *
 * @param input_dim
 * @param embed_dims
 * @param dropoutPer
 * @param output_layer
 */
class MultiLayerPerceptron(input_dim: Long, embed_dims: Seq[Long], dropoutPer: Double, output_layer: Boolean = true) extends Module {

  val layers = new ListBuffer[Module]()
//  val moduleDict = new StringAnyModuleDict()
  val moduleDict = new ModuleDictImpl()

  var output_linear: LinearImpl = new LinearImpl(input_dim, 1)
  var steps = 0
  for (embed_dim <- embed_dims) {
    var linearBnReluDropBlock = new LinearBnReluDropBlock(input_dim, embed_dim, dropoutPer, steps.toString)
    register_module(s"linearBnReluDrop_${steps}", linearBnReluDropBlock)
//    moduleDict.register_module(s"linearBnReluDrop_${steps}",linearBnReluDropBlock)
//    moduleDict.insert(s"linearBnReluDrop_${steps}", linearBnReluDropBlock.asInstanceOf[AnyModule])
    steps += 1
    layers.+=(linearBnReluDropBlock)
  }
  val mlp = new SequentialImpl(moduleDict)
  if (output_layer) {
    //    output_linear = new LinearImpl(input_dim, 1)
    register_module(s"output_linear", output_linear)
  }

  def forward(xs: Tensor): Tensor = {
    var x = xs
    val moduleIter = this.mlp.begin()
    val moduleIterEnd = this.mlp.end()
    var index = 0
    while (!moduleIter.equals(moduleIterEnd)) {
      x = moduleIter.get().asInstanceOf[LinearBnReluDropBlock].forward(x)
      //      x = mlp.get(index).forward(x)
      moduleIter.increment()
    }
    if (output_layer) {
      x = output_linear.forward(x)
    }
    x
  }
}


```

but StringAnyModuleDict  ModuleDictImpl  ModuleListImpl

**Comment by mullerhai** [comment]
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00000001862fb35b, pid=1784, tid=0x0000000000000d03
#
# JRE version: Java(TM) SE Runtime Environment (8.0_191-b12) (build 1.8.0_191-b12)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.191-b12 mixed mode bsd-amd64 compressed oops)
# Problematic frame:
# C  [libjnitorch.dylib+0x30b35b]  Java_org_bytedeco_pytorch_ModuleListImpl_insert+0x1ab
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /Users/zhanghaining/Documents/codeWorld/pytorch-scala-fm/hs_err_pid1784.log

**Comment by saudet** [comment]
Duplicate of #1283
